lib/Mojo/DOM/CSS.pm

```diff
@@ -173,18 +215,27 @@ sub _namespace {
   return !length $ns;
 }
 
+sub _parent {
+  my ($tree) = @_;
+  return $tree->[3] if $tree->[0] eq 'tag';
+  return $tree->[2] if $tree->[0] eq 'text';
```

Comment:
You seem to have misunderstood me. Now this conditional doesn't make sense. There is no else case.

Reply:
When the only option is `text` here, why do you check at all?

Reply:
The else case won't be hit unless the function is called with the `root` node, which isn't currently the case. With or without the conditional, the line is dead in its current usage.

Do you prefer a plain `return undef;` without the conditional?

Reply:
I don't know what the correct code would be, i'm just saying that the conditional that's currently there on the last line of the function is not useful.

Reply:
Personally, I find it useful for the cases to be explicitly listed, so I can verify at a glance that the function covers the cases I want it to cover.

To move one step up in the tree, there are 3 cases: If you're on a tag, the parent is entry 3; if you're on a text node, the parent is entry 2; if you're on the root node, there is no parent. 

Easy to read and verify that it does what I'd like it to do.

If you would prefer it written in another way, I can do that, but it'd probably be best you tell me how you'd like it written in that case, rather than me trying variations until something sticks. :)